lib/ansible/plugins/lookup/inventory_hostnames.py

```diff
@@ -34,41 +34,20 @@
     type: list
 """
 
-from ansible.inventory.manager import split_host_pattern, order_patterns
+from ansible.errors import AnsibleError
+from ansible.inventory.manager import InventoryManager
 from ansible.plugins.lookup import LookupBase
-from ansible.utils.helpers import deduplicate_list
 
 
 class LookupModule(LookupBase):
-
-    def get_hosts(self, variables, pattern):
-        hosts = []
-        if pattern[0] in ('!', '&'):
-            obj = pattern[1:]
-        else:
-            obj = pattern
-
-        if obj in variables['groups']:
-            hosts = variables['groups'][obj]
-        elif obj in variables['groups']['all']:
-            hosts = [obj]
-        return hosts
-
     def run(self, terms, variables=None, **kwargs):
-
-        host_list = []
-
-        for term in terms:
-            patterns = order_patterns(split_host_pattern(term))
-
-            for p in patterns:
-                that = self.get_hosts(variables, p)
-                if p.startswith("!"):
-                    host_list = [h for h in host_list if h not in that]
-                elif p.startswith("&"):
-                    host_list = [h for h in host_list if h in that]
-                else:
-                    host_list.extend(that)
-
-        # return unique list
-        return deduplicate_list(host_list)
+        manager = InventoryManager(self._loader, sources=['%s,' % variables['groups']['all'][0]])
```

Comment:
This `sources` line is a little weird at first glance, but if you provide no `sources` then you get a warning about failure to parse inventory.  This just bypasses that by feeding in an initial host.

Reply:
This could still cause a warning if a custom list of inventory plugins is enabled. Maybe InventoryManager could take a parameter to bypass it? If implicit localhost is the only inventory, the group all will be an empty list too.

Looks good to me besides that.